src/sagas/pushNotification.js

```diff
@@ -0,0 +1,440 @@
+ * @param {{ data: Object, from: string, messageId: string, sentTime: number, ttl: number }} message - Message received from wallet-service
+ * @example
+ * {
+ *   bodyLocArgs: '[\"10 T2\",\"5 T1\",\"2\"]',
+ *   bodyLocKey: 'new_transaction_received_description_with_tokens',
+ *   titleLocKey: 'new_transaction_received_title',
+ *   txId: 'txId1',
+ * }
+ * @inner
+ */
+const messageHandler = async (message) => {
+  const { data } = message;
+  if (!localization.hasKey(data.titleLocKey)) {
+    console.debug('unknown message titleLocKey', data.titleLocKey);
+    return;
+  }
+
+  if (!localization.hasKey(data.bodyLocKey)) {
+    console.debug('unknown message bodyLocKey', data.bodyLocKey);
+    return;
+  }
+
+  const bodyArgs = JSON.parse(data.bodyLocArgs);
+  const title = localization.getMessage(data.titleLocKey);
+  const body = localization.getMessage(data.bodyLocKey, bodyArgs);
+
+  notifee.displayNotification({
+    title,
+    body,
+    android: {
+      channelId: TRANSACTION_CHANNEL_ID,
+    },
+  });
+};
+
+const confirmDeviceRegistrationOnFirebase = async () => {
+  try {
+    // Make sure deviceId is registered on the FCM
+    await messaging().registerDeviceForRemoteMessages();
+  } catch (error) {
+    console.error(`Error confirming the device is registered on FCM: ${error.message}`, error);
+  }
+};
+
+const installForegroundListener = () => {
+  try {
+    // Add listeners for push notifications on foreground and background
+    messaging().onMessage(onForegroundMessage);
+  } catch (error) {
+    console.error(`Error installing foreground listener to push notification: ${error.message}`, error);
+  }
+};
+
+const createChannelIfNotExists = async () => {
+  try {
+    const hasTransactionChannel = await notifee.isChannelCreated(TRANSACTION_CHANNEL_ID);
+    if (!hasTransactionChannel) {
+      await notifee.createChannel({
+        id: TRANSACTION_CHANNEL_ID,
+        name: TRANSACTION_CHANNEL_NAME,
+      });
+    }
+  } catch (error) {
+    console.error(`Error creating channel for push notification: ${error.message}`, error);
+  }
+};
+
+/**
+ * This function is called when the wallet is initialized with success.
+ */
+export function* onAppInitialization() {
+  yield take(types.START_WALLET_SUCCESS);
+
+  const { enabled, showAmountEnabled } = STORE.getItem(pushNotificationKey.settings) || { enabled: false, showAmountEnabled: false };
+  const hasBeenEnabled = STORE.getItem(pushNotificationKey.hasBeenEnabled);
+  const enabledAt = STORE.getItem(pushNotificationKey.enabledAt);
+
+  const persistedDeviceId = STORE.getItem(pushNotificationKey.deviceId);
+  const deviceId = yield call(getDeviceId);
+  // If the deviceId is different from the persisted one, we should update it.
+  // The first time the perisistedDeviceId will be null, and the deviceId will be
+  // the one returned by getDeviceId, which gets the deviceId from FCM.
+  if (deviceId && persistedDeviceId !== deviceId) {
```

Comment:
If `deviceId` is `null`, it means the request to fetch it failed. How is the app handling this? What happens in that case?